SpatialGDK/Source/Private/Interop/SnapshotManager.cpp

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SnapshotManager.h"
+
+#include "EngineClasses/SpatialNetDriver.h"
+#include "Interop/Connection/SpatialWorkerConnection.h"
+#include "Interop/GlobalStateManager.h"
+#include "Interop/SpatialReceiver.h"
+#include "SpatialConstants.h"
+#include "Utils/SchemaUtils.h"
+
+DEFINE_LOG_CATEGORY(LogSnapshotManager);
+
+using namespace improbable;
+
+void USnapshotManager::Init(USpatialNetDriver* InNetDriver, UGlobalStateManager* InGlobalStateManager)
+{
+	NetDriver = InNetDriver;
+	Receiver = InNetDriver->Receiver;
+	GlobalStateManager = InGlobalStateManager;
```

Comment:
```suggestion
	GlobalStateManager = InNetDriver->GlobalStateManager;
```

If this isn't possible due to access could you make it possible?